platforms/chibios/boards/STEMCELL/ld/STEMCELL_tinyuf2.ld

```diff
@@ -0,0 +1,90 @@
+/*
+    ChibiOS - Copyright (C) 2006..2018 Giovanni Di Sirio
+    Copyright (C) 2022 Mega Mind
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+/*
+ * STeMCell (STM32F4x1Cx) memory setup.
+ */
+
+MEMORY
+{
+    flash0 (rx) : org = 0x08000000, len = 32k        /* Sector 0,1 - TinyUF2 bootloader */
+    flash1 (rx) : org = 0x08008000, len = 32k        /* Sector 2,3 - Emulated eeprom */
```

Comment:
Unfortunately the way the emulated EEPROM is written, it's going to be situated at the 16kB boundary instead of 32kB.
This makes it difficult to support tinyuf2, given that the bootloader binary is 14kB already and doesn't leave us with a lot of wiggle room.

Reply:
Pardon me, I did not understand this clearly. 

tinyuf2 binary size is around 28 Kb and tinyuf2 linker file uses 32kb (first two sectors)
https://github.com/adafruit/tinyuf2/blob/master/ports/stm32f4/linker/stm32f4.ld#L47

I tried to use the next two sectors of each 16KB for EEPROM.
So I defined the EEPROM_BASE_ADDRESS here appropriately:
https://github.com/qmk/qmk_firmware/pull/16287/files#diff-acf2e3a1560d0c323e6c1fce4f5625c68242220a6968c69f175e1df985d1b03eR106

So actual qmk code start from Sector 4(5th sector). Do you foresee any issues with this memory arrangement ?

Reply:
As I stated, the way that the code is written is that it's hard-coded to use the 2nd 16k page:
https://github.com/qmk/qmk_firmware/blob/edd1f33013084a79cffcfafafe7c2cbd381d6979/platforms/chibios/eeprom_stm32_defs.h#L64-L66

Defining the location in _only_ the ldscript will not change the implementation using the second page.

Reply:
The above code uses the hard coded value, only if users does not define their own `FEE_PAGE_BASE_ADDRESS `
And i defined the FEE_PAGE_BASE_ADDRESS manually here 
https://github.com/qmk/qmk_firmware/pull/16287/files#diff-acf2e3a1560d0c323e6c1fce4f5625c68242220a6968c69f175e1df985d1b03eR106

Is it not sufficient? I have tested the EEPROM config with via and rgb and did not see any issues with it.

Reply:
BTW, #16586 also attempts to enable EEPROM emulation together with tinyuf2; maybe the default value of `FEE_PAGE_BASE_ADDRESS` should depend on the bootloader. Changing `FEE_PAGE_BASE_ADDRESS` for the default case (DFU bootloader in system memory) is probably not desired, because it wastes some flash and increases the apparent firmware size (QMK does not use the DfuSe format extensions which would allow the firmware to be sparse, therefore the whole area reserved for EEPROM and whatever else would need to be flashed together with the rest of the firmware).

Reply:
I have removed the hard coded `FEE_PAGE_BASE_ADDRESS` and moved it to use WEAR_LEVELING_DRIVER with legacy backend.